File: src/ml/Models/StatisticsNumerical.py

import json

class StatisticsNumerical:
    def __init__(
        self,
        valid_count   = 0,
        na_count      = 0,
        unique_count  = 0,
        mean          = 0,
        std_deviation = 0,
        median        = 0,
        quantiles_25  = 0,
        quantiles_50  = 0,
        quantiles_75  = 0,
        min           = 0,
        max           = 0
        ) -> None:
        
        self.ValidCount   = int(valid_count)
        self.NaCount      = int(na_count)
        self.UniqueCount  = int(unique_count)
        self.Mean         = str(float(mean))
        self.StdDeviation = str(float(std_deviation))
        self.Median       = str(float(median))
        self.Quantile25   = str(float(quantiles_25))
        self.Quantile50   = str(float(quantiles_50))
        self.Quantile75   = str(float(quantiles_75))
        self.Minimum      = str(float(min))
        self.Maximum      = str(float(max))

File: src/backend/dotNet/dotNet/Models/StatisticsNumerical.cs
namespace dotNet.Models {
    public class StatisticsNumerical {
        public StatisticsNumerical(
            int validCount, int naCount, int uniqueCount, float mean, 
            float standardDeviation, float median, float quantile25, 
            float quantile50, float quantile75, float minimum, float maximum) {

            ValidCount = validCount;
            NaCount = naCount;
            UniqueCount = uniqueCount;
            Mean = mean;
            StdDeviation = standardDeviation;
            Median = median;
            Quantile25 = quantile25;
            Quantile50 = quantile50;
            Quantile75 = quantile75;
            Minimum = minimum;
            Maximum = maximum;
        }

        public static Dictionary<string, StatisticsNumerical>? Load(string data) {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Dictionary<string, StatisticsNumerical>>(data);
        }

        public int ValidCount { get; set; }
        public int NaCount { get; set; }
        public int UniqueCount { get; set; }
        public float Mean { get; set; }
        public float StdDeviation { get; set; }
        public float Median { get; set; }
        public float Quantile25 { get; set; }
        public float Quantile50 { get; set; }
        public float Quantile75 { get; set; }
        public float Minimum { get; set; }
        public float Maximum { get; set; }
    }
}
